lib/fencing/st_output.c

```diff
@@ -62,34 +100,69 @@ last_fenced_xml(pcmk__output_t *out, va_list args) {
     time_t when = va_arg(args, time_t);
 
     if (when) {
-        crm_time_t *crm_when = crm_time_new(NULL);
         xmlNodePtr node = xmlNewNode(NULL, (pcmkXmlStr) "last-fenced");
         char *buf = NULL;
 
-        crm_time_set_timet(crm_when, &when);
-        buf = crm_time_as_string(crm_when, crm_time_log_date | crm_time_log_timeofday | crm_time_log_with_timezone);
+        buf = time_t_string(when);
 
         xmlSetProp(node, (pcmkXmlStr) "target", (pcmkXmlStr) target);
         xmlSetProp(node, (pcmkXmlStr) "when", (pcmkXmlStr) buf);
 
         pcmk__xml_add_node(out, node);
-
-        crm_time_free(crm_when);
         free(buf);
     }
 
     return 0;
 }
 
+static int
+stonith_event_html(pcmk__output_t *out, va_list args) {
+    stonith_history_t *event = va_arg(args, stonith_history_t *);
+    int full_history = va_arg(args, int);
+    char *buf = NULL;
+
+    switch(event->state) {
+        case st_done:
+            buf = crm_strdup_printf("%s of %s successful: delegate=%s, client=%s, origin=%s, %s='%s'",
+                                    stonith_action_str(event->action), event->target,
+                                    event->delegate ? event->delegate : "",
+                                    event->client, event->origin,
+                                    full_history ? "completed" : "last-successful",
+                                    time_t_string(event->completed));
+            out->list_item(out, NULL, buf);
+            free(buf);
+            break;
+
+        case st_failed:
+            buf = crm_strdup_printf("%s of %s failed : delegate=%s, client=%s, origin=%s, %s='%s'",
+                                    stonith_action_str(event->action), event->target,
+                                    event->delegate ? event->delegate : "",
+                                    event->client, event->origin,
+                                    full_history ? "completed" : "last-failed",
+                                    time_t_string(event->completed));
+            out->list_item(out, NULL, buf);
```

Comment:
e.g. if name were "failed-stonith-event" we could make it red. Not that this particular case is likely to be ever used in the real world, but it's a nice capability.